src/physics/base/PhysicsInterface.hh

```diff
@@ -60,11 +60,13 @@ struct ModelGroup
  * It is allowable for this to be "false" (i.e. no materials assigned)
  * indicating that the value table doesn't apply in the context -- for
  * example, an empty ValueTable macro_xs means that the process doesn't have a
- * discrete interaction.
+ * discrete interaction. Only macro_xs ValueTables will have energy_max
+ * assigned, and only for processes that also have an energy_loss ValueTable.
```

Comment:
I think this would be an argument for moving `energy_max` to a separate struct/group outside of ValueTable -- I'm assuming you thought about this; was it just too awkward to introduce another struct? You could probably simplify by making an `ItemRange<real_type> energy_max_xs;` inside `ProcessGroup` that is either empty (if none of the processes for the particle are continuous-discrete, or if the integral treatment is disabled) or is a flattened 2D array indexed like `[material][ppid]` (so that you don't have to know the total number of materials).

Reply:
You're right, I did have this in a separate struct at one point. I debated whether it added much except a little extra code, but it was probably less awkward than what I have here, so I'll go back to that. I also like your suggestion of the flattened array in `ProcessGroup` (we'd also need to store a mapping of process ID to ppid index in that array, and the number of energy loss processes, right?), so I could do that instead, whichever you think makes more sense.